instance/models/mixins/database.py

```diff
@@ -231,3 +234,26 @@ def deprovision_mongo(self):
                 mongo.drop_database(database)
             self.mongo_provisioned = False
             self.save()
+
+
+class RabbitMQInstanceMixin(models.Model):
+    """
+    An instance that uses a RabbitMQ vhost with a set of users
+    """
+    rabbitmq_vhost = models.CharField(max_length=16, blank=True)
+    rabbitmq_provider_user = models.OneToOneField(RabbitMQUser, on_delete=models.CASCADE, blank=True)
+    rabbitmq_consumer_user = models.OneToOneField(RabbitMQUser, on_delete=models.CASCADE, blank=True)
+    rabbitmq_provisioned = models.BooleanField(default=False)
+
+    class Meta:
+        abstract = True
+
+    def provision_rabbitmq(self):
+        """
+        Create RabbitMQ vhost and users
+        """
+
+    def deprovision_rabbitmq(self):
+        """
+        Delete RabbitMQ vhost and users
+        """
```

Comment:
Sorry, I actually pushed an old version of the code, but it's fixed now. These functions are filled with something and the file has a newline.